src/main/java/slimeknights/tconstruct/library/client/RenderUtil.java

```diff
@@ -1,12 +1,342 @@
+        break;
+      case NORTH:
+        renderer.pos(matrix, x1, y1, z1).color(r, g, b, a).tex(minU, maxV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x1, y2, z1).color(r, g, b, a).tex(minU, minV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x2, y2, z1).color(r, g, b, a).tex(maxU, minV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x2, y1, z1).color(r, g, b, a).tex(maxU, maxV).lightmap(light1, light2).endVertex();
+        break;
+      case SOUTH:
+        renderer.pos(matrix, x1, y1, z2).color(r, g, b, a).tex(maxU, maxV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x2, y1, z2).color(r, g, b, a).tex(minU, maxV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x2, y2, z2).color(r, g, b, a).tex(minU, minV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x1, y2, z2).color(r, g, b, a).tex(maxU, minV).lightmap(light1, light2).endVertex();
+        break;
+      case WEST:
+        renderer.pos(matrix, x1, y1, z1).color(r, g, b, a).tex(maxU, maxV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x1, y1, z2).color(r, g, b, a).tex(minU, maxV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x1, y2, z2).color(r, g, b, a).tex(minU, minV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x1, y2, z1).color(r, g, b, a).tex(maxU, minV).lightmap(light1, light2).endVertex();
+        break;
+      case EAST:
+        renderer.pos(matrix, x2, y1, z1).color(r, g, b, a).tex(minU, maxV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x2, y2, z1).color(r, g, b, a).tex(minU, minV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x2, y2, z2).color(r, g, b, a).tex(maxU, minV).lightmap(light1, light2).endVertex();
+        renderer.pos(matrix, x2, y1, z2).color(r, g, b, a).tex(maxU, maxV).lightmap(light1, light2).endVertex();
+        break;
+    }
+  }
+
+  /**
+   * Similar to putTexturedQuad, except its only for upwards quads and a rotation is specified
+   */
+  public static void putARotatedQuad(IVertexBuilder renderer, TextureAtlasSprite sprite, double x, double y, double z, double w, double d, Direction rotation,
+    int color, int brightness, boolean flowing) {
+    int l1 = brightness >> 0x10 & 0xFFFF;
+    int l2 = brightness & 0xFFFF;
+
+    int a = color >> 24 & 0xFF;
+    int r = color >> 16 & 0xFF;
+    int g = color >> 8 & 0xFF;
+    int b = color & 0xFF;
+
+    putARotatedQuad(renderer, sprite, x, y, z, w, d, rotation, r, g, b, a, l1, l2, flowing);
+  }
+
+  /**
+   * Similar to putTexturedQuad, except its only for upwards quads and a rotation is specified
+   */
+  public static void putARotatedQuad(IVertexBuilder renderer, TextureAtlasSprite sprite, double x, double y, double z, double w, double d, Direction rotation,
```

Comment:
We no longer need the dumb names for these methods, go back to `putTexturedQuad` and `putRotatedQuad`. The only reason they were renamed is because a mod decided for some dumb reason to hack into our rendering code with unsafe code, causing crashes for months before we got tired of dealing with the reports.

Reply:
@shedaniel ^

Reply:
actually I'll do this one. I just don't know the javadoc @shedanial